### PR TITLE
Added orientation, compression and id_section as TGA save keyword arguments

### LINKED ISSUE #3226: Sanitize TGA Image.info
Currently, the TGA image plugin sets `Image.info` "compression" to "tga_rle" on `load()`, while `save()` takes "rle" (bool) option. Both methods should use the same option for consistency. It probably makes sense to keep "rle" as TGA format doesn't support other compression methods, but "compression" may be more consistent with BMP and TIFF plugins. Neither of the options is documented, so there is no danger of breaking backward compatibility.

Also, it's not very clear whether `save()`  method should "inherit" info like TIFF and PNG plugins do:

https://github.com/python-pillow/Pillow/blob/4407cb65079a7d1150277e3b9a144996f56357c9/src/PIL/TiffImagePlugin.py#L1399-L1400

Currently, TGA plugin only inherits "orientation" but doesn't allow to specify it as a keyword to `save()`, and "id_section" is ignored altogether.

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #3226

### CODE REVIEW

File: src/PIL/TgaImagePlugin.py [hunk 1]
```diff
@@ -166,11 +174,12 @@ def _save(im, fp, filename):
...
         flags = 0
 
-    orientation = im.info.get("orientation", -1)
+    orientation = im.encoderinfo.get("orientation",
+                                     im.info.get("orientation", -1))
     if orientation > 0:
         flags = flags | 0x20
 
-    fp.write(b"\000" +
+    fp.write(o8(idlen) +
```
Comment (danpla) [inline src/PIL/TgaImagePlugin.py]: There is a small problem: o8 is `bytes((i & 255,))`, which will not raise an exception like `struct.pack()`, making it possible for `fp.write()` below to write `id_section` bigger than 255 bytes, resulting in a broken image.

Comment (radarhere) [reply src/PIL/TgaImagePlugin.py]: Okay, I've added a commit to trim `id_section`.

File: src/PIL/TgaImagePlugin.py [hunk 2]
```diff
@@ -151,11 +151,19 @@ def _save(im, fp, filename):
     except KeyError:
         raise IOError("cannot write mode %s as TGA" % im.mode)
 
-    rle = im.encoderinfo.get("rle", False)
-
+    if "rle" in im.encoderinfo:
```
[the remark this answers is not in the document]
Comment (radarhere) [reply src/PIL/TgaImagePlugin.py]: Yeah. I'm just not keen to break the backwards compatibility of this one. If this PR is merged, you're welcome to submit a separate PR for that.

File: src/PIL/TgaImagePlugin.py [hunk 3]
```diff
@@ -151,11 +153,23 @@ def _save(im, fp, filename):
...
+                                         im.info.get("compression"))
+        rle = compression == "tga_rle"
     if rle:
         imagetype += 8
 
+    id_section = im.encoderinfo.get("id_section",
+                                    im.info.get("id_section", ""))
+    idlen = len(id_section)
+    if idlen > 255:
+        idlen = 255
```
Comment (hugovk) [inline src/PIL/TgaImagePlugin.py]: Let's `snake_case` this variable.

File: src/PIL/TgaImagePlugin.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hugovk) [reply src/PIL/TgaImagePlugin.py]: Sounds sensible to keep it, we've no idea what code might be using `im.encoderinfo["rle"]`.

We'd need deprecation warnings first before removing.

File: src/PIL/TgaImagePlugin.py (hunk 3 above)
Comment (radarhere) [reply src/PIL/TgaImagePlugin.py]: Okay, I've added a commit. Note that the commit also changes the `idlen` variable in `_open`, for consistency.

**Pull request merged**